platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java

```diff
@@ -0,0 +1,142 @@
+  }
+
+  /**
+   * Adds an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param account The account being observed.
+   * @param expiring true if the account can be expired. false, otherwise.
+   */
+  public void upsert(String account, Boolean expiring) throws ValueValidationException {
+    if (account != null && expiring != null) {
+      upsert(new ObservingAccount(account, Instant.now(), expiring));
+    }
+  }
+
+  /**
+   * Add an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param observingAccount The account being observed.
+   */
+  public void upsert(ObservingAccount observingAccount) throws ValueValidationException {
+    if (observingAccount != null) {
+      ObservingAccount existingAccount = allAccounts.get(observingAccount.account);
+      if (existingAccount == null) {
+        allAccounts.put(observingAccount.account, observingAccount);
+        // update the database
+        store.upsert(observingAccount.account, observingAccount.lastObserved);
+      } else {
+        if (!observingAccount.canExpire.equals(existingAccount.canExpire))
+          throw new ValueValidationException(
+              String.format(
+                  "The expiring flag cannot be modified. Account=[%s]", observingAccount.account));
+        existingAccount.account = observingAccount.account;
+        existingAccount.lastObserved = observingAccount.lastObserved;
+      }
+    }
+  }
+
+  /**
+   * Gets the list of observed accounts.
+   *
+   * @return The list of observed accounts.
+   */
+  public List<ObservingAccount> getAccounts() {
+    return new ArrayList<>(allAccounts.values());
+  }
+
+  /**
+   * Look up if the account is being observed. If the account is being observed, the lastObserved
+   * timestamp of the observing account will be updated.
+   *
+   * @param account The account to be checked.
+   * @return true if the account is being observed. false, otherwise.
+   */
+  public boolean lookupAndUpdate(String account) {
+    ObservingAccount acct = allAccounts.get(account);
+    if (acct == null) return false;
+    acct.lastObserved = Instant.now();
+    return true;
+  }
+
+  /**
+   * Evict expired accounts
+   *
+   * @param maxIdleTime evict all accounts that are older than maxAge
+   */
+  public void evict(Duration maxIdleTime) {
+    for (ObservingAccount acct : getAccounts()) {
+      if (!acct.canExpire) continue;
+
+      Duration idleTime = Duration.between(Instant.now(), acct.lastObserved).abs();
+      if (idleTime.compareTo(maxIdleTime) > 0) {
+        allAccounts.remove(acct.account);
+        store.delete(acct.account);
+      }
+    }
+  }
+
+  @AllArgsConstructor
+  public static class ObservingAccount {
+    String account;
+    Instant lastObserved;
+    Boolean canExpire;
+  }
+
+  Duration getEvictPeriod() {
+    return Duration.of(1, HOURS);
+  }
+
+  Duration getEvictMaxIdleTime() {
+    return Duration.of(7, DAYS);
+  }
```

Comment:
👀  I know this is still arbitrary, but I'm more comfortable having a longer period than 1 WEEK. Can you at least bump it to 2 weeks, since this is not configurable for now?

I'd appreciate if you could either make this as an optional configuration in this PR or create an issue to address it later.

Reply:
The issue is created. https://github.com/stellar/java-stellar-anchor-sdk/issues/492

The max idle time is increased to 1 MONTH and flush period is set to 5 minutes.

Reply:
Cool, thank you!